till_korten/devbio-napari_cluster/readme.md

```diff
@@ -0,0 +1,214 @@
+* your workflow needs a graphical user interface other than standard jupyter notebook stuff
+
+### See also
+* [ZIH HPC Documentation](https://doc.zih.tu-dresden.de/)
+
+## Step 1: Get access to the ZIH cluster
+Fill out a project proposal and submit it to the compute center as [explained in their documentation](https://tu-dresden.de/zih/hochleistungsrechnen/zugang/projektantrag?set_language=en).
+
+## Step 2: Start a Jupyter session on the ZIH cluster
+You will be greeted with the TUD login screen. Log in with your ZIH user name and password:
+
+<img src="images/1_login.png" width="500" />
+
+
+Afterwards, you should see a single button `Start My Server`. Click on it:
+
+<img src="images/2_start_server.png" width="500" />
+
+Now you get to configure the computing node you want your session to run on. Switch to the Advanced configuration by clicking the button `Advanced`. Then you should see something like the image below. 
+
+Start by choosing a preset (click on 1). You should choose a GPU node preset (2). You can choose either a n Ampere A100 or a Tesla K80 card. Choose according to the your needs and the current utilization (the bars above the preset chooser) (the A100 are on `alpha`, the K80 are on `gpu2`) If the partition is very full, you may have to wait a long time or not get a session at all. Choose the number of CPUs and memory per cpu as needed. Note that the memory is per CPU, so if you choose more CPUs, you get more memory.
+
+Once you are happy with your configuration, click the orange button `Spawn` at the very bottom.
+
+<img src="images/3_configure_node.png" width="500" />
+
+You will now see a wait bar. Do not worry if it does not move, this bar is always at 50%. It usually takes 2-5 min to get a node.
+
+<img src="images/4_wait.png" width="500" />
+
+## Step 3: Open a terminal
+
+Open a terminal by clicking on file (1 in the image below) -> new (2) -> Terminal (3)
+
+<img src="images/5_open_terminal.png" width="500" />
+
+## Step 4: Install a custom jupyter kernel for your user
+
+In the terminal, execute the following code:
+
+```bash
+git clone https://gitlab.mn.tu-dresden.de/bia-pol/singularity-devbio-napari.git
+```
+You will be asked for a user name and password. Enter the following:
+
+User name: `git_pull`
+
+Password: `Bc6Yg8k5MyZzURNuWChR`
+
+Once the download has finished, enter the following:
+
+```bash
+cd singularity-devbio-napari
+./install.sh <version>
+```
+
+replace `<version>` with the version of our devbio-napari environment you want to use. At the time of writing, the latest version was `v0.1.5`. Always specify a version (or a tag name you got from us in case we created a custom environment for you). This will ensure that your code is always executed in the same environment and your data analysis is reproducible. Note down the version you use in your lab book!
```

Comment:
Since the repository is private, we cannot post a link (more precisely, users would not be able to open it).

This is another argument for making the repository public.

We cannot go our usual route and publish it on github, because the CI workflow is gitlab-specific. Gitlab.com might work, but I don't know what their space restrictions are for the container registry.
The easiest would be to make the project public as-is, but last time I tried that setting was greyed out.